src/errno.rs

```diff
@@ -52,28 +53,64 @@ pub fn errno() -> i32 {
     }
 }
 
-macro_rules! impl_errno {
-    ($errno:ty) => {
-        impl $errno {
-            pub fn last() -> Errno {
-                super::last()
-            }
+impl Errno {
+    pub fn last() -> Self {
+        last()
+    }
 
-            pub fn desc(self) -> &'static str {
-                super::desc(self)
-            }
+    pub fn desc(self) -> &'static str {
+        desc(self)
+    }
 
-            pub fn from_i32(err: i32) -> Errno {
-                from_i32(err)
-            }
+    pub fn from_i32(err: i32) -> Errno {
+        from_i32(err)
+    }
 
-            pub unsafe fn clear() -> () {
-                super::clear()
-            }
+    pub unsafe fn clear() -> () {
+        clear()
+    }
+
+    pub fn result<S: ErrnoSentinel + PartialEq<S>>(value: S) -> Result<S> {
```

Comment:
Makes sense. I'd like it to work for both, but the design I had in mind requires method default type parameters, [which aren't ready yet](https://github.com/rust-lang/rust/pull/30724).